plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotPageSourceProvider.java

```diff
@@ -25,39 +30,61 @@
 import io.trino.spi.connector.ConnectorTableHandle;
 import io.trino.spi.connector.ConnectorTransactionHandle;
 import io.trino.spi.connector.DynamicFilter;
+import org.apache.pinot.common.proto.Server;
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.common.utils.DataTable;
+import org.apache.pinot.core.common.datatable.DataTableFactory;
+import org.apache.pinot.core.transport.ServerInstance;
+import org.apache.pinot.spi.utils.CommonConstants;
 
 import javax.inject.Inject;
 
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
 import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.IntStream;
 
+import static io.trino.plugin.pinot.PinotErrorCode.PINOT_EXCEPTION;
 import static io.trino.plugin.pinot.query.DynamicTablePqlExtractor.extractPql;
 import static io.trino.plugin.pinot.query.PinotQueryBuilder.generatePql;
+import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
 
 public class PinotPageSourceProvider
         implements ConnectorPageSourceProvider
 {
     private final PinotQueryClient pinotQueryClient;
+    private final PinotGrpcClient pinotGrpcClient;
     private final PinotClient clusterInfoFetcher;
     private final int limitForSegmentQueries;
     private final int limitForBrokerQueries;
     private final int estimatedNonNumericColumnSize;
     private final long targetSegmentPageSizeBytes;
+    private final boolean grpcEnabled;
 
     @Inject
     public PinotPageSourceProvider(
             PinotConfig pinotConfig,
             PinotClient clusterInfoFetcher,
-            PinotQueryClient pinotQueryClient)
+            PinotQueryClient pinotQueryClient,
+            PinotGrpcClient pinotGrpcClient)
```

Comment:
Given that the grpc setting is global to the server, Instead of passing both, it'd be better for these to implement a common interface and pass the appropriate one depending how the connector was initialized.